src/components/stringclass.js

```diff
@@ -0,0 +1,249 @@
+
+function wordCheck(name){
+    /*Check if the string is empty */
+    if(!name || name === ""){
+        return "Please add a string"; 
+    } else{return true;}
+}
+
+function hasVowels(name){
+    /*Checks if the string contains vowels*/
+    if(wordCheck(name)=== true){
+        let vowels = ['a','e','i','o','u'];
+        let nameSplit = name.split("");
+        for(let word of nameSplit){
+            for(let vowel of vowels){
+                if(word === vowel){
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+    
+}
+
+function toUpper(name){
+    /*Converts a string to uppercase */
+     if(wordCheck(name)=== true){
+        let newString = [];
+        for(let word of name.split(" ")){
+            let resultVar = "";
+            for(let i = 0; i < word.length; i++) {
+                resultVar += String.fromCharCode(word.charCodeAt(i) & 223);
+            }
+            newString.push(resultVar);
+        }
+        if(newString.join(" ") === name){
+            return "String is already in uppercase";
+        }else {
+            return newString.join(" ");
+        }
+    }
+    
+}
+
+function toLower(name) {
+    /*Converts a string to lowercase */
+    if(wordCheck(name)=== true){
+        let newString = [];
+        for(let word of name.split(" ")){
+            let resultVar = "";
+            for (let i = 0; i < word.length; i++) {
+                let code = word.charCodeAt(i);
+                if (code > 64 && code < 91) {
+                    resultVar += String.fromCharCode(code + 32);
+                } else {
+                    resultVar += word.charAt(i);
+                }
+            }
+            newString.push(resultVar);
+        }
+        if(newString.join(" ") === name){
+            return "String already in lowercase";
+        } else{
+            return newString.join(" ");
+        }
+    }
+    
+}
+
+function ucFirst(name){
```

Comment:
Consider using descriptive variable names ie `firstToUpper`. One of the advantages of doing this is that it makes your code self describing and reduces the amount of commenting done in the code.

Reply:
`ucFirst` was the specific method name in the CP2 document.